### PR TITLE
Update tests to follow developer guidelines.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Moved tests to match their corresponding test src files.
- Updated test namespaces to match their corresponding test src files.

#782 

/cc @pranavkm

### CODE REVIEW

File: test/Microsoft.AspNetCore.Razor.Test/Editor/RazorEditorParserTest.cs [hunk 1]
```diff
@@ -1,256 +0,0 @@
...
-using Microsoft.AspNetCore.Razor.Test.Framework;
-using Microsoft.AspNetCore.Razor.Test.Generator;
-using Microsoft.AspNetCore.Razor.Test.Utils;
-using Microsoft.AspNetCore.Razor.Text;
-using Microsoft.AspNetCore.Testing;
-using Xunit;
-
-namespace Microsoft.AspNetCore.Razor.Test.Editor
-{
-    public class RazorEditorParserTest
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Test/Editor/RazorEditorParserTest.cs]: There were two test classes testing the same class, moved this into that one.

File: test/Microsoft.AspNetCore.Razor.Test/Parser/CSharpDirectivesTest.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@
 using Microsoft.AspNetCore.Razor.Test.Framework;
 using Xunit;
 
-namespace Microsoft.AspNetCore.Razor.Test.Parser.CSharp
+namespace Microsoft.AspNetCore.Razor.Parser
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Razor.Test/Parser/CSharpDirectivesTest.cs]: Remove the using

Comment (pranavkm) [inline test/Microsoft.AspNetCore.Razor.Test/Parser/CSharpDirectivesTest.cs]: Similarly for a bunch of the files that follow.

Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Test/Parser/CSharpDirectivesTest.cs]: Must have been a bad rebase on my part. Did this prior :cry:

**Comment by NTaylorMullen** [comment]
:up: :date:

File: test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/CSharpRazorCodeGeneratorTest.cs [hunk 3]
```diff
@@ -4,13 +4,13 @@
 using System;
 using System.Collections.Generic;
 using Microsoft.AspNetCore.Razor.Chunks.Generators;
-using Microsoft.AspNetCore.Razor.CodeGenerators;
 using Microsoft.AspNetCore.Razor.Parser.SyntaxTree;
+using Microsoft.AspNetCore.Razor.Test.CodeGenerators;
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/CSharpRazorCodeGeneratorTest.cs]: Should the types in the namespace be moved to `Microsoft.AspNetCore.Razor.CodeGenerators`?

File: test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/CSharpTagHelperRenderingTest.cs [hunk 4]
```diff
@@ -13,7 +13,7 @@
 using Microsoft.AspNetCore.Testing.xunit;
 using Xunit;
 
-namespace Microsoft.AspNetCore.Razor.Test.Generator
+namespace Microsoft.AspNetCore.Razor.Test.CodeGenerators
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/CSharpTagHelperRenderingTest.cs]: `Microsoft.AspNetCore.Razor.CodeGenerators`

File: test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/CSharpRazorCodeGeneratorTest.cs (hunk 3 above)
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/CSharpRazorCodeGeneratorTest.cs]: I moved the classes that test a specific implementation. However, there's infrastructure bits that are only test related. Do we typically move infrastructure bits into their corresponding src namespaces.

Comment (pranavkm) [inline test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/CSharpRazorCodeGeneratorTest.cs]: We do have some precedence for it. For instance https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNetCore.Mvc.Core.Test/TestFeatureProvider.cs#L11

File: src/Microsoft.AspNetCore.Razor/Tokenizer/Symbols/Internal/HtmlSymbolExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Razor/Tokenizer/Symbols/Internal/HtmlSymbolExtensions.cs]: Remove usings

File: test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/CSharpTagHelperRenderingUnitTest.cs [hunk 6]
```diff
@@ -4,13 +4,13 @@
...
 using Microsoft.AspNetCore.Razor.CodeGenerators.Visitors;
+using Microsoft.AspNetCore.Razor.Compilation.TagHelpers;
 using Microsoft.AspNetCore.Razor.Parser.SyntaxTree;
 using Microsoft.AspNetCore.Razor.Parser.TagHelpers;
 using Microsoft.AspNetCore.Razor.TagHelpers;
-using Microsoft.AspNetCore.Razor.Compilation.TagHelpers;
 using Xunit;
 
-namespace Microsoft.AspNetCore.Razor.Test.Generator
+namespace Microsoft.AspNetCore.Razor.Test.CodeGenerators
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/CSharpTagHelperRenderingUnitTest.cs]: Couple more of these that could be moved to `Microsoft.AspNetCore.Razor.CodeGenerators`

**Comment by pranavkm** [comment]
:shipit:

**Comment by NTaylorMullen** [comment]
Closing this for now so I can point it to dev.

**Pull request closed without merging**